schema/schema_builder.hh

```diff
@@ -291,8 +291,10 @@ public:
     // Equivalent to with(cp).build()
     schema_ptr build(compact_storage cp);
 
-    schema_ptr build();
+    schema_ptr build() &;
```

Comment:
could also be: const &.

Reply:
I thought about it. Then the newly introduced `build(raw_schema& s)` should also be `const`, but in this case we have

```c++
schema_ptr build() && {
    return build(_raw);
}

schema_ptr build(raw_schema& s) const {
    ... // modifies s here
}
```

So const method is called with a non-const reference on this's member. It's not supposed to change "this", but it will do it.

Reply:
> Could make raw_schema a value parameter. Then it `build() &&` will first detach _raw from this.

It's 616 bytes in size

> Or, even simpler, `static build(raw_schema&)`.

It still needs more data from schema_builder -- `_version`, `_compact_storage` and `_view_info`, so it's not much simpler :(